sympy/solvers/tests/test_solveset.py

```diff
@@ -1701,6 +1712,10 @@ def test_nonlinsolve_complex():
         (ImageSet(Lambda(n, 2*n*I*pi + log(sin(2))), S.Integers), 2)})
 
 
+def test_nonlinsolve_radical():
+    assert nonlinsolve([sqrt(y) - x - z, y - 1], [x, y, z]) == {(1 - z, 1, z)}
```

Comment:
On master, `nonlinsolve([sqrt(y) - x - z, y - 1], [x, y, z])` gives `{(1 - z, 1, z), (-z - 1, 1, z)}`, where the second solution is incorrect. The incorrect solution is not returned on this PR because `nonlinsolve` now checks if each solution satisfies all equations that were modified using `unrad` (the first equation in this case) and only returns the solutions that satisfy all such equations.